### DISCUSSION TITLE
Classic Explorer Treeview - High DPI issue in Windows 10

### DISCUSSION THREAD

**Discussion by Alex-302** [ramensoftware/windhawk-mods#584] [General]
Hello @CyprinusCarpio

### Issue

I have an issue when using 150% system scale - line overlapping and visible default collapse icon.

`+` icon is not a problem - it is possible to disable it, but small lines issue makes this great mod non usable :(

![image](https://github.com/ramensoftware/windhawk-mods/assets/8361299/2e73034f-ace1-404d-8750-c744417b19aa)

Screenshot of Explorer with 150% when the mod is ON and OFF:

<details>
<summary>Details</summary>

![image](https://github.com/ramensoftware/windhawk-mods/assets/8361299/3c440aea-0809-4d76-9d0e-4f5aa801e706)

</details>

### Config
Windows 10 x64 (22H2 16045.4170)
27" UDH display with 150% system scale
Windhawk mods: Classic Explorer Treeview, Taskbar Thumbnail Reorder

**Comment by CyprinusCarpio** [comment]
Hey!

My mod works best when it's used under the Classic theme. Contrary to popular belief, it has not been removed from windows, it can be enabled using several techniques on modern versions including up to 11 23H2. Classic theme is the primary concern for this mod.

Secondary concern are various Win XP themes based on the Basic non-DWM theme.

Various themes after Win XP are a afterthought, they were not tested on because the treeview had a fundamental change after XP, so my mod is not accurate on those themes anyways.

The +/- button option is meant to be used on the Classic theme. My mod has to draw it's own +/- buttons because for some reason micro$oft removed them on the classic theme after Win7, even though the classic theme was not 'usable' after win7. Why spend time explicitly removing small, cosmetic things from a deprecated feature when there's nothing to gain from doing so? No idea.

The overlapping lines issue is a byproduct of yet another workaround of stupid micro$oft ideas. In their infinite 'wisdom' they have added a `iIntegral` integer value for every tree item, which multiplies the height of individual items by it's value. Why not just set the item height directly, you know, as it's been done on just about every windows version other than 10? On win10 it is set to 3 for every item, and even better, for some reason this value being set to anything other than 1 causes the item icons to be obstructed by white rectangles when the dotted lines are enabled! So, to achieve a look consistent with Win 2000 (and allow the dotted lines to be drawn without glitches), I set this value to 1 for every item in a hook `NSCSubClassTreeWndProcHook`. If you fork the mod and comment out the if statements in `NSCSubClassTreeWndProcHook` for the messages `TVM_SETITEMHEIGHT`, `TVM_INSERTITEM` and `TVM_SETITEM` the issue with lines overlapping should be solved for you.

**Reply by Alex-302** [reply]
Thanks. Design of modern Win is pain in back.

Result:

No horizontal line, but it much better when expanded few directories.

<details>
<summary>Details</summary>

![image](https://github.com/ramensoftware/windhawk-mods/assets/8361299/e1044804-41b0-48e8-99f6-b87f46dcc68b)

</details>

or without buttons

<details>
<summary>Details</summary>

![image](https://github.com/ramensoftware/windhawk-mods/assets/8361299/58fc8c6b-6854-4702-88e9-0012068fa13f)


</details>


Correct change?

```
        // else if(uMsg == TVM_SETITEMHEIGHT) //Set item height to the accurate value
        // {
        //     // wParam = 16;
        // }
        // else if (uMsg == TVM_INSERTITEM)
        //     //Ensure that iIntegral is set to 1 for every item to achieve accurate item height
        //     LPTVINSERTSTRUCT lpis = (LPTVINSERTSTRUCT)lParam;
        //     lpis->itemex.iIntegral = 1;
        // else if (uMsg == TVM_SETITEM) // Quick access item workaround
        //     LPTVITEMEXW lptvi = (LPTVITEMEXW)lParam;
        //     if (lptvi->mask & TVIF_INTEGRAL)
        //     {
        //         lptvi->iIntegral = 1;
        //     }
```
 Same result when only this code is commented out

```
        // else if (uMsg == TVM_INSERTITEM)
        // {
        //     //Ensure that iIntegral is set to 1 for every item to achieve accurate item height
        //     LPTVINSERTSTRUCT lpis = (LPTVINSERTSTRUCT)lParam;
        //     lpis->itemex.iIntegral = 1;
        // }
```

**Reply by CyprinusCarpio** [reply]
You propably should also disable the dotted lines, they won't look good with the modern Aero theme, much less with the scaling being set to anything other than 100%. I don't think it's possible to have the dotted lines in a Aero themed treeview and not have them look like shit (cue regedit key treeview).

In any case, if you think that modern windows design is bullshit, I encourage you to check out the modding scene at [winclassic](https://winclassic.boards.net/), we've made significant strides in the past year or so, it's now possible to have a more Win2000-like look than it was on Vista 17 years ago.

**Reply by Alex-302** [reply]
Thanks. Will test winclassic
